src/geometry/manifold/instances/units_of_normed_algebra.lean

```diff
@@ -62,4 +172,54 @@ variables {𝕜 : Type*} [nontrivially_normed_field 𝕜] [normed_algebra 𝕜 R
 instance : smooth_manifold_with_corners 𝓘(𝕜, R) Rˣ :=
 open_embedding_coe.singleton_smooth_manifold_with_corners 𝓘(𝕜, R)
 
+lemma cont_mdiff_coe {m : with_top ℕ} : cont_mdiff 𝓘(𝕜, R) 𝓘(𝕜, R) m (coe : Rˣ → R) :=
+cont_mdiff_open_embedding 𝓘(𝕜, R) units.open_embedding_coe
+
+/-- Multiplication of units of a complete normed ring is a smooth map between manifolds.
+
+It suffices to show that `coe ∘ mul : Rˣ × Rˣ → R` is smooth. This function is equal to the usual
+ring multiplication composed with the embedding from `Rˣ × Rˣ` to `R × R`, and we know each of these
+factors is smooth. -/
+lemma smooth_mul :
+  smooth (𝓘(𝕜, R).prod 𝓘(𝕜, R)) 𝓘(𝕜, R) (λ (p : Rˣ × Rˣ), p.fst * p.snd) :=
```

Comment:
Please "inline" this lemma and the next (that is, instead of having them as stand-along lemmas, move the proofs to the single place at line 222/223 where they are invoked).  After the Lie group construction, people who need this fact will get it by using the `smooth_mul` fact for a general Lie group.